packages/client/lib/CredentialRequestClient.ts

```diff
@@ -119,9 +124,33 @@ export class CredentialRequestClient {
     debug(`Acquiring credential(s) from: ${credentialEndpoint}`);
     debug(`request\n: ${JSON.stringify(request, null, 2)}`);
     const requestToken: string = this.credentialRequestOpts.token;
-    let response = (await post(credentialEndpoint, JSON.stringify(request), { bearerToken: requestToken })) as OpenIDResponse<CredentialResponse> & {
+
+    let dPoP = createDPoPOpts ? await createDPoP(getCreateDPoPOptions(createDPoPOpts, credentialEndpoint, { accessToken: requestToken })) : undefined;
+
+    let response = (await post(credentialEndpoint, JSON.stringify(request), {
+      bearerToken: requestToken,
+      customHeaders: { ...(dPoP && { dpop: dPoP }) },
+    })) as OpenIDResponse<CredentialResponse> & {
       access_token: string;
     };
+
+    let nextDPoPNonce = createDPoPOpts?.jwtPayloadProps.nonce;
+    const retryWithNonce = dPoPShouldRetryResourceRequestWithNonce(response);
+    if (retryWithNonce.ok && createDPoPOpts) {
```

Comment:
What if retryWithNonce.false && createDPopOpts?

Reply:
That's valid. For example, the request already succeeded without the nonce, in which case we don't need to retry it.

I agree that createDPoPOpts is a bit annoying here. I included it because otherwise, we need to throw an Error in case retryWithNonce. OK === true and createDPoPOpts is not set, which is a combination that is not possible. Would you prefer to throw?